### DISCUSSION TITLE
Share reading

### DISCUSSION THREAD

**Discussion by Onslivion** [coinfoundry/miningcore#1194] [Q&A] [answered]
Trying to see how I could implement shares into my front end. Pending shares for an ethash pool are odd;

<img width="514" alt="Screen Shot 2022-03-06 at 18 43 36" src="https://user-images.githubusercontent.com/25300749/156950006-d365c6d7-3427-4580-8f15-26ba329ff5bf.png">

What does this number mean? It appears like the shares are submitted as if they were share * difficulty, but I'm not sure.
How would I go about using this number?

Current configuration is PPLNS, with;
```
            "varDiff": {
              "minDiff": 0.3,
              "maxDiff": 1.87,
              "targetTime": 15,
              "retargetTime": 90,
              "variancePercent": 30
            }
``` 
Planning on narrowing this in production.

**Comment by oliverw** [comment, marked as answer]
It's exactly that. An accumulation of the difficulties of a miner's submitted shares.

**Reply by benjatd** [reply]
Hi Oliver. 

How do you get Miningcore to work with NiceHash AsicBoost?
The pool is AsicBoost compatible, but not with NiceHash AsicBoost

**Comment by Onslivion** [comment]
How would I get the count for the total amount of shares then through API?

**Reply by oliverw** [reply]
> How would I get the count for the total amount of shares then through API?

The count is a meaningless value without factoring in the difficulty. This is why is implemented the way it is.

**Reply by Onslivion** [reply]
> > How would I get the count for the total amount of shares then through API?
> 
> The count is a meaningless value without factoring in the difficulty. This is why is implemented the way it is.

Makes sense. Thanks.

One more question - when would this hash expire? It’s been nearly 18 hours with no pending shares expiring

**Reply by oliverw** [reply]
The expire once a block is found and the shares get paid accordingly.
